### ISSUE TITLE
paramiko 2.4.1 fails when current version of gssapi is installed.

### ISSUE THREAD

**Issue by Thomasillo** [paramiko/paramiko#1256]
```
$ pip list | grep gssapi
gssapi                    1.5.1  
$ pip list | grep paramiko
paramiko                  2.4.1   
$ python
...
>>> import paramiko
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Thomas/.local/miniconda3/envs/4cast/lib/python3.6/site-packages/paramiko/__init__.py", line 22, in <module>
    from paramiko.transport import SecurityOptions, Transport
  File "/Users/Thomas/.local/miniconda3/envs/4cast/lib/python3.6/site-packages/paramiko/transport.py", line 38, in <module>
    from paramiko.auth_handler import AuthHandler
  File "/Users/Thomas/.local/miniconda3/envs/4cast/lib/python3.6/site-packages/paramiko/auth_handler.py", line 48, in <module>
    from paramiko.ssh_gss import GSSAuth, GSS_EXCEPTIONS
  File "/Users/Thomas/.local/miniconda3/envs/4cast/lib/python3.6/site-packages/paramiko/ssh_gss.py", line 54, in <module>
    GSS_EXCEPTIONS = (gssapi.GSSException,)
AttributeError: module 'gssapi' has no attribute 'GSSException'
```

**Comment by Thomasillo** [comment]
Imho, the following line in `tests/util.py` is to blame.
`needs_gssapi = pytest.mark.skipif(True, reason="No GSSAPI to test")`

Seriously, the test, testing usage of the gssapi is skipped by default?!?!

**Comment by Thomasillo** [comment]
I would like to give a try fixing this. Can I be allowed to open a branch?

**Comment by ploxiln** [comment]
see also #1166 , #861 , #584

**Comment by bitprophet** [comment]
Thanks @ploxiln !

**Comment by detrout** [comment]
Hello, 

Over in Debian we were wondering which, if any, of the patches are considered acceptable by the paramiko developers?

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892860

**Comment by ploxiln** [comment]
https://github.com/paramiko/paramiko/pull/1311 was adjusted and merged and included in release 2.6.0

**Comment by detrout** [comment]
Thank you @ploxiln !
